pkg/il/compiler/compiler_test.go

```diff
@@ -388,6 +388,20 @@ end
 `,
 	},
 	{
+		expr: `ar["b"]`,
```

Comment:
won't this lead to the issue that originally caused us to make `""` => `nil` ?  what happens for: `ar["b"] | "other"`?

Reply:
Depending on the bag contents, ar["b"] | "other" will evaluate to:
bag {} => "other"
bag { ar:{} } => "other"
bag { ar:{ "b": ""} } => ""

Reply:
Do we have tests for those cases? If not, can we add them?

Reply:
Yes we do. I've also added one more, just in case, in an additional changelist within the same PR.